### ISSUE TITLE
Both clearnet and onion (and i2p) node

### ISSUE THREAD

**Issue by POMATu** [RetroShare/RetroShare#1539]
Hi,

I am having clearnet node. So I can connect to onion friends through tor. But as far as I understand they don't connect to me at all.

I am proposing feature request, so my clearnet node can have also onion and i2p addresses at the same time so connection can be both ways.

Like when I post my cert anywhere and there will include my ip, onion domain and i2p domain.

**Comment by G10h4ck** [comment]
This is a long time wanted feature, it requires a lot of work in retroshare internal and broad understanding of them, if someone want to implement it I can give guidance

**Comment by POMATu** [comment]
I don't have time and experience to implement that either. Now I get the point, due to extreme decentralization it actually works without that and it's stable, so maybe fine to leave that as is until somebody would want to implement that.

**Comment by matthias-hoste** [comment]
@G10h4ck I'd implement it, guidance would be handy since I never deeply browsed the retroshare source

**Comment by G10h4ck** [comment]
> @G10h4ck I'd implement it, guidance would be handy since I never deeply browsed the retroshare source

So this part of the code is specially intricate, and I have been looking into it many times and got to the conclusion that it need a major rewrite to support multiple listeners at same time, it is alla around pqissllistener.
Probably better you start getting acquainted to the code by solving simpler issues before we could discuss on how such a thing could be implemented, but you are definitely welcome to try :+1:

**Comment by csoler** [comment]
Exactly my opinion too. We definitely need good programmers, but this task is even hard for us although we've been familiar with this code for years.

**Comment by matthias-hoste** [comment]
Yeah you are probably right. I'd love to contribute to this amazing project. I'll scavenge the issue list a little :)
-------- Original Message --------
